### PR TITLE
[DevDependency] Update ESLint style rules

### PR DESCRIPTION
**Pull request by dorian451**
This PR  replaces the [deprecated ESLint style rules](https://eslint.org/blog/2023/10/deprecating-formatting-rules/) in our ESLint config to the newer, maintained ones by [ESLint Stylistic](eslint.style). Since the previous rules were broken and not working properly, this PR also updates  any non-conformant files to be conformant with our style rules.

This PR also changes our [arrow-parens](https://eslint.style/rules/js/arrow-parens) style rule to "always".

### CODE REVIEW

File: site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js [hunk 1]
```diff
@@ -44,21 +43,20 @@ const checkStudentsInCache = () => {
     for (const user_id of all_user_ids) {
         // Gradeable # of late days used should be empty
         cy.get(`[USER_ID="${user_id}"] > [id="Late Allowed Homework"]`)
-            .then((cell) => expect(cell.text().trim()).to.equal(''));
+            .then(cell => expect(cell.text().trim()).to.equal(''));
 
         // Remaining late days isnt known, should be empty
         cy.get(`[USER_ID="${user_id}"] > [ld_id="Late Days Remaining"]`)
-            .then((cell) => expect(cell.text().trim()).to.equal(''));
```
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: Is there a reason that some of the then() instances use parenthesis and some of them don't? (similar with .each())

Comment (dorian451) [reply site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: The reason is that those were auto fixed, and eslint's linting rules don't really care about whether there are parentheses or not. I can look into enabling an additional rule that would force parentheses or not, but so far I've tried to make the config as close as possible to the original.

Comment (IDzyre) [reply site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: The ones that had parentheses were changed to not, and the ones that didn't have any were changed to have them.

Comment (IDzyre) [reply site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: I think 'always' would make everything more consistent, but I'd ask @williamjallen for his opinion.

**Comment by IDzyre** [comment]
The ansible test is failing because of a 'launchpad' issue, see https://x.com/launchpadstatus

File: site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js (hunk 1 above)
Comment (dorian451) [reply site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: Personally I like 'always' more too. I'll wait for him to comment.

**Comment by williamjallen** [comment]
@dorian451 It looks like you have some merge conflicts now.  Please also update the PR description.

File: site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js (hunk 1 above)
Comment (williamjallen) [reply site/cypress/e2e/Cypress-Feature/late_days_cache.spec.js:51]: "Always" seems reasonable to me.

**Comment by IDzyre** [comment]
These PRs should probably be priority to review and merge, as there are a lot of PRs that have Cypress tests, that currently are not being linted, and will just increase the work and merge conflicts with this PR

**Pull request merged**